### DISCUSSION TITLE
"OnUse" StatusEffects always give medical skill when healing

### DISCUSSION THREAD

**Discussion by UnhappyWithMyUsername** [FakeFishGames/Barotrauma#14381] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [X] My issue happened while using mods.

### What happened?

Despite setting MedicalSkillGain="0" for custom Thirst affliction, reducing the affliction still gives a massive amount of medical skill.

There is currently no way to prevent type="OnUse" StatusEffects from increasing skills.

### Solution:
StatusEffects should either only increase skill on "OnSuccess"/"OnFailure"
_or_
introduce a new attribute like SkillGain="false" (true by default) to disable any skill gain from this StatusEffect.

### Here's a video:

https://github.com/user-attachments/assets/4d6192c6-1e00-4991-9365-e49b51e59446

Here's the relevant part of the item code:
```xml
<!-- Regular drinking -->
<StatusEffect type="OnSecondaryUse" target="This" Condition="99" setvalue="true" comparison="And">
  <Conditional Condition="equals 100" />
  <Conditional InWater="false" />
  <Sound file="%ModDir:2954998725%/Content/Sounds/Items/Drinks/OpenCan.ogg" loop="false" range="200" volume="1.5" />
</StatusEffect>
<StatusEffect type="OnUse" target="This" Condition="-12.0">
  <Conditional InWater="false" />
</StatusEffect>
<StatusEffect type="OnUse" target="Character" comparison="And">
  <Conditional InWater="false" />
  <ReduceAffliction identifier="drunk" amount="1.5" />
  <ReduceAffliction identifier="drunknodebuffs" amount="1.5" />
  <ReduceAffliction identifier="hallucinating" amount="1.5" />
  <Affliction identifier="haste" amount="12.0" />
  <Sound file="%ModDir:2954998725%/Content/Sounds/Items/Drinks/Drinking.ogg" loop="true" range="200" volume="5.0" />
</StatusEffect>
<StatusEffect type="OnUse" target="Character" comparison="And">
  <Conditional thirst="gt 1.75" />
  <Conditional InWater="false" />
  <ReduceAffliction type="thirst" amount="1.75" />
</StatusEffect>
<StatusEffect type="OnUse" target="Character" comparison="And">
  <Conditional bladder="gte 0.01" />
  <Conditional InWater="false" />
  <Affliction identifier="fillbladder" amount="1.75" />
</StatusEffect>
```

And the affliction with MedicalSkillGain="0.0":
```xml
<Affliction name="Thirst"
  identifier="thirst"
  description="You feel refreshed."
  type="thirst"
  causeofdeathdescription="Died of thirst"
  selfcauseofdeathdescription="You have died of thirst."
  limbspecific="false"
  indicatorlimb="Torso"
  hideiconafterdelay="true"
  iconcolors="0,120,200,255;180,0,0,255"
  activationthreshold="0"
  showinhealthscannerthreshold="0"
  treatmentthreshold="20"
  maxstrength="100"
  healableinmedicalclinic="false"
  MedicalSkillGain="0.0">
  <Description
    textidentifier="afflictiondescription.he-thirst.self"
    target="Self"
    minstrength="30"
    maxstrength="80" />
  <Description
    textidentifier="afflictiondescription.he-thirst.lethal.self"
    target="Self"
    minstrength="80"
    maxstrength="100" />
  <Description
    textidentifier="afflictiondescription.he-thirst.other.low"
    target="OtherCharacter"
    minstrength="0"
    maxstrength="30" />
  <Description
    textidentifier="afflictiondescription.he-thirst.other"
    target="OtherCharacter"
    minstrength="30"
    maxstrength="80" />
  <Description
    textidentifier="afflictiondescription.he-thirst.lethal.other"
    target="OtherCharacter"
    minstrength="80"
    maxstrength="100" />
  <!-- No effects at this point -->
  <Effect minstrength="0" maxstrength="50" strengthchange="0" />
  <!-- Thirst decreases vitality and speed slightly -->
  <Effect minstrength="50" maxstrength="65" strengthchange="0" 
    minvitalitydecrease="0"
    maxvitalitydecrease="15"
    minskillmultiplier="1.0"
    maxskillmultiplier="0.8" />
  <Effect minstrength="65" maxstrength="80"
    minvitalitydecrease="15"
    maxvitalitydecrease="30"
    minskillmultiplier="0.8"
    maxskillmultiplier="0.6"
    minscreendistort="0"
    maxscreendistort="0"
    minradialdistort="0"
    maxradialdistort="0.1"
    strengthchange="0"
    minspeedmultiplier="1.0"
    maxspeedmultiplier="0.8" />
  <!-- Thirst starts to get lethal -->
  <Effect minstrength="80" maxstrength="100"
    minvitalitydecrease="30"
    maxvitalitydecrease="100"
    minskillmultiplier="0.6"
    maxskillmultiplier="0.2"
    minscreendistort="0"
    maxscreendistort="1"
    minscreenblur="0"
    maxscreenblur="3.0"
    minradialdistort="0.1"
    maxradialdistort="1.0"
    strengthchange="0"
    minspeedmultiplier="0.8"
    maxspeedmultiplier="0.5">
    <StatusEffect target="Character" >
      <Conditional thirst="gte 100" />
      <Affliction identifier="he-dieofthirst" strength="0.1" />
    </StatusEffect>
  </Effect>
  <icon texture="%ModDir:2954998725%/Content/UI/HE-IconsAtlas.png" sourcerect="128,128,128,128" origin="0,0" />
</Affliction>
```

### Reproduction steps

1. Create an item with shootable="true" requireaimtouse="true" and an "OnUse" StatusEffect that heals some afflictions in a Holdable component.
2. Make sure the afflictions you configured in the item are present on your character.
3. Use the item and watch your skill gain increase. (Increases faster, the higher the affliction values are)
4. Realize there's no way to prevent medical skill gain for OnUse StatusEffects.

ALTERNATIVELY:
1. Download Hungry Europans and load into the campaign/sub-editor
2. Open console and type
3. giveaffliction thirst 90
4. spawnitem he-water inventory

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Happens in both single player and multiplayer

### Version

v1.5.9.1 (Summer Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by Regalis11** [comment]
The skill gain seems to be coming from the Hyperactivity buff the item applies, not from reducing the Thirst affliction. There seems to be a bug with the way skills are increased when applying buffs over time, which is why the Hyperactivity is giving more skill than it should (see https://github.com/FakeFishGames/Barotrauma/discussions/14379).

**Reply by UnhappyWithMyUsername** [reply]
But if that's the case, why does it not apply to characters that aren't thirsty? Or does it come from the buff being applied, but the skill gain only happens if the character doesn't have full vitality?

**Reply by Regalis11** [reply]
In my tests, I could not spot any difference no matter if the character is thirsty or not. The hyperactivity seems to increase the skill level rapidly in both cases, but the Thirst doesn't seem to have any effect (reducing it gives no skill gains).

https://github.com/user-attachments/assets/d7e4c480-f2c3-4b73-b381-9baaf812711d
